feature/projectile

keliboi opened the pull request:
Merge feature branch for ID4 submission requirements

lecegues committed:
Updated `users.js` to extract JWT_SECRET from environment variable

lecegues committed:
Updated `docker-compose.yaml` to use .env variables

- especially to account for private but constant JWToken

lecegues committed:
Temporary change to workflow file to accomodate for `JWT_SECRET` .env variable

- Will be changed in a later patch to be private

funmise committed:
added a baselayout page for other pages to extend

funmise committed:
fixed the issue and clean up a bit

lecegues committed:
Removed POSTGRES_PASSWORD env variable from docker-compose file

- POSTGRES_PASSWORD could not reference .env file `DB_PASS` variable
- Therefore, its better to add POSTGRES_PASSWORD in .env file

lecegues committed:
Created unit tests for authentication feature

lecegues committed:
Created authentication middleware to authenticate admin-only actions

funmise committed:
commented out a function, didn't do what I thought it did

funmise committed:
corrected small error i made in importing the logger

lecegues committed:
Removed unnecessary quotes that are causing parsing errors

funmise committed:
added empty files for future use, cleaned up, and made basic setup for future development

m-jewel committed:
Implemented PeriodModal so user can edit period attribute.

m-jewel committed:
Implemented CultureModal so user can edit culture's name attribute. Also modified PeriodModal where I deleted <br/> for good UI formatting.

m-jewel committed:
Implemented BaseShapeModal so user is able to edit base shape attrubute. Also edited CulturalModal and PeriodModal because they were calling wrong endpoints.

m-jewel committed:
Implemented BladeShapeModal so users can edit the Blade Shape attribute.

m-jewel committed:
Implemented HaftingShape so users can modify Hafting Shape attributes.

m-jewel committed:
Implemented CrossSectionModal so users can edit Cross Section attribute.

Commit:
Implemented a adding projectile point to database using user interface, and template for editable attributes for a projectile point

Commit:
Updated fetching of all projectile points to include site and artifactType to group be able to group projectile points by site

m-jewel committed:
* Edited ProjectileModal.jsx: Implemented functionality of PeriodModal in the Add Projectile Modal(ProjectileModal.jsx) so the user can add, edit, and delete periods when/while they are creating a new Projectile. Also included some comments that relates to PeriodModal's functionalities.
* Edited PeriodModal.jsx: Added doctrings, comments, assertions, and loggers so it would be easy to understand for others.
* Edited backend/app.js: Modified the cors option so it includes "PUT" method so I could update periods during editing menu.
* Created Unit test to test the basiuc functionalites of PeriodModal.jsx

m-jewel committed:
Edit comment dividers so it is easy for me to find which sections are for specific modals.

Commit:
Created siteID variable for code readability

Commit:
Added siteID as prop from parent component, to be used to filter projectile points by siteID

Commit:
Added separate states for add projectile and view projectile

m-jewel committed:
* Created unit tests for CultureModal.jsx to test basic functionality
* Modifed ProjectileModal.jsx: Implemented CultureModal functionalities to the Add Projectile Modul so the user can add, edit, and delete a Culture attribute
* Modified CultureModal.jsx: Included logger and documentations so teammates can understand the code

Commit:
Added projectile point page as modal, to view projectile point details

Commit:
Merge branch 'feature/projectile' of ssh://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-4 into feature/projectile

m-jewel committed:
* Created frontend/tests/baseShapeModal.unit.test.jsx for BaseShapeModal unit testing to verify it's basic functionalites.
* Modified frontend/src/components/BaseShapeModal.jsx: Added some comments and doctrings for easy understanding.
* Modified frontend/src/components/CultureModal.jsx: Made the docstring more detailed so the parameters are easy to understand / let people know what parameters are needed.

Commit:
Added Site component unit test script

Commit:
Merge branch 'feature/projectile' of ssh://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-4 into feature/projectile

m-jewel committed:
*Modified frontend/src/components/ProjectileModal.jsx: Integrated CrossSectionModal.jsx to ProjectileModal so user can add, edit, and delete a Cross Section attribute
* Modified frontend/src/components/CrossSectionModal.jsx: Added more detailed doctrings for easier understanding of the pre-conditions and post-conditions

Commit:
Merge branch 'feature/projectile' of ssh://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-4 into feature/projectile

m-jewel committed:
* Modified frontend/src/components/BladeShapeModal.jsx: Added more detailed doctring to let developers know what are the pre and post conditions

* Modifed frontend/src/components/ProjectileModal.jsx: Integrated BladeShapeModul.jsx to ProjectileModul.jsx so users are able to add, edit, and delete Blade Shape attributes while creating a new projectile

m-jewel committed:
* Modified and renamed frontend/src/components/HaftingShapeModal.jsx: Added some comments to explain what function does and also renamed it by adding Modal in the end to retain consistency.

* Modifed frontend/src/components/ProjectileModal.jsx: Integrated HaftingShape.jsx so the user can add, edit, and delete a haftingShape attribut

* Some note: When adding a new haftingShape, console log says that input was a success and it does get added in the database. However, console log also says that there was an error when adding the Hafting shape due to type error (line 40 in HaftingShapeModal.jsx.

Commit:
Updated HTTP request from fetch() to axios for development standards

Commit:
Added adding a projectile point functionality to Jewel's attribute modals

funmise committed:
fixed bug where adding a new projectile did not automatically refresh the list

Commit:
Integrated Jewel's attribute modals/dropdowns with adding a projectile to database, for complete core adding functionality

Commit:
Removed unnecessary comments for cleaner code

Commit:
Fixed hafting shape modal bug

Commit:
Removed unnecessary comments for clean code

Commit:
Cleanup user interface for better spacing

Commit:
Adjusted form elements spacing for cleaner layout

Commit:
Cleanup user interface for better spacing

Commit:
Updated sites list and projectiles list layout for better spacing and cleaner look

lecegues committed:
Merge branch 'patch/issue-92' into feature/projectile

Commit:
Fixed region menu UI when adding a site, implemented adding, editing, and deleting materials

m-jewel committed:
Refactored importing logger to have name of log instead for consistency name calling with other frontend components.

Commit:
added test cases for rendering the componet correctly, interacting with database and done the testing through UI as well. but since dropdown menu is not properly adding the region in the form so one test case has faied. the actual functionality of the feature is working

m-jewel committed:
Added docstrings from frontend/src/components files starting from letter R to U to follow team best practice and so others can understand what is happening in the code.

Commit:
Added docstrings and removed unnecessary comments for code readability

m-jewel committed:
Fixed problem where Sidebar components
did not load due to refactoring done earilier

Commit:
git conflit resolved

m-jewel committed:
Added docstrings to backend/src/routes files: artifact, artifactTypes, and baseShapes. Aditionally, incorporated logger and assertions for easy testing and debugging.

Commit:
Added docstrings, removed unnecessary comments, and removed unused variables for code readability

m-jewel committed:
Added doctrings for route files, bladeShapes, catalogues, and crossSection so it is easy to understand what its pre, post, and parameters. Also added status code, logger, and assertions for easy testing.

m-jewel committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/projectile' into feature/projectile

m-jewel committed:
Fixed logger import naming that causes backend crash because of wrong spelling during refactoring.

m-jewel committed:
Fixed Catalogue name import naming that causes backend crash because of wrong spelling during refactoring.

Commit:
some test cases are not behaving the same way since there was some change pulled

Commit:
Merge branch 'feature/projectile' of https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-4 into feature/projectile

Commit:
integration test added for adding site

m-jewel committed:
Added doctstrings to backend route files: cultures and haftingShape so others know the pre, pos, and parameter conditions. Additionally used logger and assertions for easier testing.

Commit:
Copied unit tests from ID2

m-jewel committed:
Fixed catalogue issue that is causing error where Base catalogue was fetched instead of Default.

m-jewel committed:
Added basic unit testing for other modals to ensure it components render correctly, edit and delete attributes asociated.

Commit:
Clean up unnecessary files to ready for pull request for ID4

Commit:
Added catalogue.unit.test.jsx from ID3 for project testability

Commit:
Updated view projectile unit test from ID3 due to functionality updates/changes and for project testability

Commit:
Fixed linting errors for code readability

Commit:
Fixed prettier formatting errors for code readability

Commit:
Fixed linting errors in backend/ folder for code readability

Commit:
integration test for the adding projectile points

lecegues committed:
Added missing multi-level logging and docstrings in `routes`

- to improve readaibility and testability

lecegues committed:
Fixed smoke test to be more specific;

- Smoke test was detecting two similar elements to press so the test failed

Commit:
Merge branch 'feature/projectile' of https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-4 into feature/projectile

Commit:
integration test added for adding site

lecegues committed:
Merge branch 'develop' into feature/projectile

lecegues committed:
Rollup Error: added package changes to fix

funmise reviewed (commented):
need to fix rollup issue. might need to add it as optional dependency

lecegues committed:
Rollup Error: fixed by updating package files

xix695 committed:
pre-populated drop-down menu integration test

m-jewel committed:
Deleted unused variables that is causing failure to merge / in pull request.

m-jewel committed:
Deleted unused variables that is causing failure to merge / in pull request.

m-jewel committed:
Deleted unused variables that is causing failure to merge / in pull request.

m-jewel committed:
Ran Prettier as it is causing pull request merge fail.

m-jewel committed:
Attempt to address PeriodModal integ test fails (ISSUE #112). Pushing this commit so my tester can access PeriodModal.jsx changes.

m-jewel committed:
Fixed linting error that prevents pipe from running the integ tests which is needed to see to know what errors/issues still need fixing.

m-jewel committed:
Renamed expected text label to match the new texts when PeriodMOdal was edited to implemnt input verifications.

lecegues committed:
Added missing parenthesis causing the backend to crash

- Also removed unnecessary duplicate file in `config`

Commit:
Removed adding the <MoreHorizIcon> component to selected region from dropdown, for easy element tagging for automated testing

Commit:
Fixed initializing regionID from dropdown selection for submitting new sites, and switched console.logs to utilize logger

m-jewel committed:
Addressed error message consistency where in one modal, the error mesage was within the modal itself. But, on MaterialModal, the error message shows up as a pop up alert.

m-jewel committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/projectile' into feature/projectile

xix695 committed:
update MaterialModal integration test

Commit:
integration test for adding sites and adding projectile is working on the loacal system

Commit:
Added labelId html attribute to region dropdown for easy tagging in automated tests

Commit:
Added id html attribute to region dropdown for easy tagging in automated tests

Commit:
Removed unnecessary test file for feature branch to avoid merge conflicts with other branches

lecegues committed:
Added a health check to make sure backend is running before running requests

lecegues committed:
Added unit, integration, coverage testing & reporting in backend

lecegues committed:
Added unit, integration, coverage testing & reporting in frontend

lecegues committed:
Added testing reports to .gitignore & created a testing-script for smoke tests

lecegues committed:
Added `test-report` directory to ESLint & Prettier ignore lists

- Fixed Linting and Prettier formatting errors

lecegues committed:
Removed unnecessary artifacts for frontend/backend folders.

- To be replaced with test document artifacts instead

lecegues committed:
Added steps to the workflow to upload frontend/backend unit test reports

lecegues committed:
Moved all tests to last step to upload all testing artifacts at once, rather than being separated

lecegues committed:
Changed step names & added coverage testing steps

lecegues committed:
Added a health check route and step to the workflow to make sure backend/database is ready before making connections/queries

lecegues committed:
Merge branch 'patch/issue-113' into CURRENT feature/projectile

- this is to access the test documents & new workflow

m-jewel committed:
- error key constraints are being outputted before because the test case is trying to add a a site without selecting a region intentionally, and the moment the test case fires an event to click the "Add" button, it will complain about the key constraints. The reason why this test case pass is it asserts that this site does not exist in the database which is true.

- So, to deal with the error constraint message itself, I modified the SiteModal so when a user clicks "Add" with empty region, it will display error to the user that a Region is required. In the test case, I added 2 lines where its expecting this Text to be displayed. Therefore, error key constraints in the terminal output is gone now.

m-jewel committed:
- Modified MaterialModal.jsx because it was exporting function CultureModal instead of MaterialModal

lecegues committed:
Increased timeout of a `site.integ` test & made sure to add pre-requisite data before running the test

lecegues committed:
Modifications to CI Workflow to increase test timeouts and ensure that tests continue upon error to generate test documents

lecegues committed:
Changed axios default adapter to http rahter than xmlhttp which could be causing errors

lecegues committed:
Removed the addition of `continue-on-error` causing steps to pass in the pipeline even if they had failed tests

lecegues committed:
Added a more verbose output on backend health check to debug connectivity issues better

lecegues committed:
Changed health check url to `regions` to see if the endpoint is accessible in general

lecegues committed:
Revert past commit & changed health check URL back to previous state

lecegues committed:
Changed default address to `127.0.0.1:3000` due to concerns of ViTest resolving `localhost` to ipv6 addresses

lecegues committed:
Hardcoded the ipv4 address to some tests to debug if the ipv4-ipv6 is the problem

lecegues committed:
Hardcoded last failing test to determine if the problem lies in ipv4-ipv6 addresses

lecegues committed:
Changed more related functions to debug ipv4-ipv6 suspected bug

lecegues committed:
Created a global axios configuration file to fix issue #115 & replaced all axios imports

- "http" will default connect to 127.0.0.1:3000 (localhost)
- can be modified by future users to change their port
- additionally, will prevent resolving to ipv6 addresses
- Also revamped all files that referenced `axios` to use this custom config file instead

funmise commented on frontend/tests/Login.integ.test.js:
this file should not be deleted.

keliboi replied:
Login.integ.test.js added back to branch

keliboi replied:
File was removed again as it is non-functional for this branch, it will be added through a merge with another branch in develop

Commit:
Added back Login integration test that was added through a merge with develop branch, test is not needed in current feature branch so it was temporarily removed

funmise commented on frontend/tests/setup.js:
```diff
@@ -2,6 +2,9 @@ import { expect, afterEach } from "vitest";
 import "@testing-library/jest-dom";
+import axios from "axios";
+axios.defaults.adapter = "http";
+axios.defaults.baseURL = "http://127.0.0.1:3000";
```
line 7 not needed. It's redundant.

keliboi replied:
Line removed as requested

funmise reviewed (changes requested):
Going forward, the integration tests should be made more rigorous to ensure that the request is actually carried out and not just sent i.e . further communication with backend.

Commit:
Line removed due to redundancy with line 6, as requested in pull request review

Commit:
Ran prettier fix to pass pipeline formatting standards

Commit:
Removed non-functional test that will be added through merge with another branch

funmise reviewed (approved):
looks good now

lecegues reviewed (approved):
Updated changes from Babz' review looks good. Good usage of multi-level logging, assertions, and docstrings (compared to past pull requests).
